generators/generator-defaults.js

```diff
@@ -16,28 +16,32 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-const constants = require('./generator-constants');
+const defaultApplicationOptions = require('../jdl/jhipster/default-application-options');
+const applicationOptions = require('../jdl/jhipster/application-options');
+const { MONOLITH } = require('../jdl/jhipster/application-types');
+const binaryOptions = require('../jdl/jhipster/binary-options');
 
-const ANGULAR = constants.SUPPORTED_CLIENT_FRAMEWORKS.ANGULAR;
+const optionNames = applicationOptions.OptionNames;
+const defaultNewApplicationOptions = defaultApplicationOptions.getConfigForApplicationType();
+const defaultMonolithOptions = defaultApplicationOptions.getConfigForApplicationType(MONOLITH);
 
 /** Required config for prompts to be skipped */
 const appRequiredConfig = {
-    applicationType: 'monolith',
+    applicationType: defaultMonolithOptions[optionNames.APPLICATION_TYPE],
 };
 
 const appDefaultConfig = {
     ...appRequiredConfig,
-    skipClient: false,
-    skipServer: false,
-    skipUserManagement: false,
+    skipClient: defaultNewApplicationOptions[optionNames.SKIP_CLIENT],
+    skipServer: defaultNewApplicationOptions[optionNames.SKIP_SERVER],
+    skipUserManagement: defaultMonolithOptions[optionNames.SKIP_USER_MANAGEMENT],
     skipCheckLengthOfIdentifier: false,
     skipFakeData: false,
-    withGeneratedFlag: false,
```

Comment:
Why you removed `withGeneratedFlag`?

Reply:
looks like a oopsie

Reply:
Looks like a mistake on my side

Reply:
It's not needed.
It's used once and directly from config.